src/pkg/control/options.go

```diff
@@ -48,44 +47,50 @@ func Defaults() Options {
 }
 
 // ---------------------------------------------------------------------------
-// Restore Item Collision Policy
+// Restore Configuration
 // ---------------------------------------------------------------------------
 
-// CollisionPolicy describes how the datalayer behaves in case of a collision.
-type CollisionPolicy int
-
-//go:generate stringer -type=CollisionPolicy
 const (
-	Unknown CollisionPolicy = iota
-	Copy
-	Skip
-	Replace
+	defaultRestoreLocation = "Corso_Restore_"
 )
 
-// ---------------------------------------------------------------------------
-// Restore Destination
-// ---------------------------------------------------------------------------
+// CollisionPolicy describes how the datalayer behaves in case of a collision.
+type CollisionPolicy string
 
 const (
-	defaultRestoreLocation = "Corso_Restore_"
+	Unknown CollisionPolicy = ""
+	Skip    CollisionPolicy = "skip"
+	Copy    CollisionPolicy = "copy"
+	Replace CollisionPolicy = "replace"
 )
 
-// RestoreDestination is a POD that contains an override of the resource owner
-// to restore data under and the name of the root of the restored container
-// hierarchy.
-type RestoreDestination struct {
-	// ResourceOwnerOverride overrides the default resource owner to restore to.
-	// If it is not populated items should be restored under the previous resource
-	// owner of the item.
-	ResourceOwnerOverride string
-	// ContainerName is the name of the root of the restored container hierarchy.
-	// This field must be populated for a restore.
-	ContainerName string
+// RestoreConfig contains
+type RestoreConfig struct {
```

Comment:
what's the long-term plan for this? Based on the struct it looks like we kind of plan to always pass this down to the restore consumer. However I still feel there's a benefit to separating the logic of generating the location something is restored at (assuming no collisions) from the logic that actually restores items/handles collisions

Reply:
also, not strictly related to this PR or this chunk of code, but we should probably make a validation function for the restore code to check these fields. Otherwise we could pass bad state down into the restore code